core/src/main/java/com/linecorp/armeria/common/logging/RequestScopedMdc.java

```diff
@@ -119,7 +120,13 @@ public final class RequestScopedMdc {
             final Field mdcAdapterField = MDC.class.getDeclaredField("mdcAdapter");
             mdcAdapterField.setAccessible(true);
             oldAdapter = (MDCAdapter) mdcAdapterField.get(null);
-            mdcAdapterField.set(null, new Adapter(oldAdapter));
+            final MDCAdapter newAdapter = new Adapter(oldAdapter);
+            mdcAdapterField.set(null, newAdapter);
+
+            if ("ch.qos.logback.classic.util.LogbackMDCAdapter".equals(oldAdapter.getClass().getName())) {
+                // Since Logback 1.4.8, LoggingEvent does not directly use MDC.getMDCAdapter().
+                DelegatingLogbackMDCAdapter.maybeUpdateMdcAdapter(newAdapter);
+            }
         } catch (Throwable t) {
             oldAdapter = null;
             logger.warn(ERROR_MESSAGE, t);
```

Comment:
Question) I think there weren't any exceptions flagged because we always add `logback` in our test runtime dependencies.

If there are users who don't include `logback` dependencies at all, wouldn't a `ClassNotFoundException` be invoked and printed here?

Reply:
The error message may be useful to detect some changes in SLFJS’ MDC implementation.
`DelegatingLogbackMDCAdapter` must not raise `ClassNotFoundExcetion` because the existence of LogbackMDCAdapter` is checked before .